### ISSUE TITLE
Clarifications related to 4.3.9 Headings and sections

### ISSUE THREAD

**Issue by rehierl** [w3c/html#912]
I am trying to implement the outline algorithm and found a few things that could
use some minor improvements:

### [4.3.9 Headings and sections](https://w3c.github.io/html/sections.html#headings-and-sections)

**(1)** Subsequent headings of equal or higher rank start new (implied) sections

More precisely, the following sentence: Subsequent headings of equal or higher
rank start new (implied) sections, headings of lower rank start implied subsections
that are part of the previous one.

The second part (headings of lower rank) makes it quite clear to which these
subsections belong.

It would help to make the first part (headings of equal or higher rank) just as
clear by mentioning, that these new (implied) sections are part of the previous
section's parent section. (At least as long as I am not mistaken).

**(2)** 4.3.9 mentions implied sections quite a lot.

It might help to mention which elements are used to create normal/explicit sections.
Probably those that result from sectioning content or sectioning root elements?

Example 24 does mention "Thud (heading of explicit 'section' section)" in 1.3

**(3)** Example 25

The section element with heading h3-Color should have the same indent as the
previous section element with heading h3-Sweet.

It is structurally sound, but a bit irritating without the proper indent.

### [4.3.9.1 Creating an outline](https://w3c.github.io/html/sections.html#creating-an-outline)

**(4)** tree order

This documentation needs some clarification on what the term 'tree order' means.
It might help to add some pseudo-code for clarification:

```
visitElement(element)
  onEnter(element)
  child = element.firstChild
  while(child != null)
    visitElement(child)
    child = child.nextSibling
  onExit(element)
```

**(5)** a section can contain any number of further nested sections

It might be more accurate to replace the word **sections** with the word **sub**sections.

**(6)** current outline target

It looks like the value named 'current outline target' refers to the current
outline's owner element. So essentially you use two different words for the same
thing: 'target' and 'owner'.

It might help to rename 'current outline target' to 'current outline owner'.

The word 'target' irritated me for quite some time.

--

**EDIT** - looks like I can't count to 6 ...

**Comment by stevefaulkner** [comment]
Thanks @rehierl will review, in what way are you implementing the outline?

**Comment by rehierl** [comment]
The short answer would be:

In JavaScript

The more verbose answer is:

For a Node.js project of mine I started to use [h5o](https://github.com/h5o/h5o-js/), but soon had to figure out that it doesn't document what you actually get when the algorithm has finished. I can call the result's 'asHTML()' function, but that is not what I need. I need to be able understand and traverse the outline myself.

Furthermore, h5o's algorithm is a pedantic step-by-step implementation, which doesn't help to understand what the algorithm is actually supposed to do. So I started to implement it myself.

Unfortunately, I had to realize that the algorithm's definition itself leaves quite a few detailed questions unanswered. So if you guys are willing to improve the definition, I might be pestering you about it in the near future :-)
